main.tf

```
@@ -1,6 +1,14 @@
 locals {
   enabled                                   = module.this.enabled
   ipv6_egress_only_internet_gateway_enabled = local.enabled && var.ipv6_egress_only_internet_gateway_enabled
+  additional_cidr_blocks_map = local.enabled ? { for v in var.additional_cidr_blocks : v => {
+    ipv4_cidr_block     = v
+    ipv4_ipam_pool_id   = null
+    ipv4_netmask_length = null
+  } } : {}
```

Just for readability

```suggestion
  additional_cidr_blocks_map = local.enabled ? {
    for cidr_block in var.additional_cidr_blocks :
    cidr_block => {
      ipv4_cidr_block     = cidr_block
      ipv4_ipam_pool_id   = null
      ipv4_netmask_length = null
    }
  } : {}
```